### PR TITLE
Point cloud 2d identify followup fixes

### PR DESCRIPTION
**Pull request by nyalldawson**
@NEDJIMAbelgacem here's some misc fixes and optimisations I found while testing the new point cloud identify functionality. Do you mind taking a look over?

### CODE REVIEW

**Comment by NEDJIMAbelgacem** [comment]
Other that minor comment, looks very good to me, thanks @nyalldawson

**Comment by nyalldawson** [comment]
@NEDJIMAbelgacem  Which comment?

File: src/core/pointcloud/qgspointclouddataprovider.h [hunk 1]
```diff
@@ -343,7 +343,7 @@ class CORE_EXPORT QgsPointCloudDataProvider: public QgsDataProvider
     void indexGenerationStateChanged( PointCloudIndexGenerationState state );
 
   private:
-    QVector<IndexedPointCloudNode> traverseTree( const QgsPointCloudIndex *pc, IndexedPointCloudNode n, float maxError, float nodeError, const QgsGeometry &extentGeometry, const QgsDoubleRange extentZRange );
+    QVector<IndexedPointCloudNode> traverseTree( const QgsPointCloudIndex *pc, IndexedPointCloudNode n, double maxError, double nodeError, const QgsGeometry &extentGeometry, const QgsDoubleRange &extentZRange );
 };
 
 #endif // QGSMESHDATAPROVIDER_H
```
Comment (NEDJIMAbelgacem) [inline src/core/pointcloud/qgspointclouddataprovider.h:349]: Why does it say QGSMESHDATAPROVIDER_H here ?

**Comment by NEDJIMAbelgacem** [comment]
Sorry I was supposed to submit the review not just comment it (stayed as pending)

File: src/core/pointcloud/qgspointclouddataprovider.h (hunk 1 above)
Comment (nyalldawson) [reply src/core/pointcloud/qgspointclouddataprovider.h:349]: I suspect a copy and paste remnant from when this file was first added

**Comment by nyalldawson** [comment]
> I was supposed to submit the review not just comment it 

I do that all the time!

**Review by wonder-sk — APPROVED** [review]

**Pull request merged**